Code/Framework/AzCore/AzCore/Memory/IAllocator.h

```diff
@@ -95,70 +57,178 @@ namespace AZ
+            "Allocators should do best effort to keep the ptr at the same address")
+        size_type Resize([[maybe_unused]] pointer ptr, [[maybe_unused]] size_type newSize)
+        {
+            return 0;
+        }
+
+        /// Realloc an allocate memory memory block. Similar to Resize except it will move the memory block if needed. Return NULL if
+        /// realloc is not supported or run out of memory.
+        //AZ_DEPRECATED_MESSAGE("Use reallocate instead, which matches the STD interface")
+        pointer ReAllocate(pointer ptr, size_type newSize, size_type newAlignment)
+        {
+            return reallocate(ptr, newSize, newAlignment);
+        }
 
-        // Every system allocator is required to provide name this is how
-        // it will be registered with the allocator manager.
-        virtual const char* GetName() const = 0;
-        virtual const char* GetDescription() const = 0;
+        ///
+        /// Returns allocation size for given address. 0 if the address doesn't belong to the allocator.
+        //AZ_DEPRECATED_MESSAGE("Use get_allocated_size instead, which matches the STD interface")
+        size_type AllocationSize([[maybe_unused]] pointer ptr)
+        {
+            return get_allocated_size(ptr);
+        }
+
+        virtual size_type NumAllocatedBytes() const
+        {
+            return 0;
+        }
+
+        /// Returns the capacity of the Allocator in bytes. If the return value is 0 the Capacity is undefined (usually depends on another
+        /// allocator)
+        //AZ_DEPRECATED_MESSAGE("Use max_size instead, which matches the STD interface")
+        size_type Capacity() const
+        {
+            return max_size();
+        }
 
-        /// Returns the schema
-        AZ_FORCE_INLINE IAllocatorSchema* GetSchema() const { return m_schema; };
+        /// Returns max allocation size if possible. If not returned value is 0
+        AZ_DEPRECATED_MESSAGE("Unused and not really useful")
+        size_type GetMaxAllocationSize() const
+        {
+            return 0;
+        }
+
+        /// Returns the maximum contiguous allocation size of a single allocation
+        //AZ_DEPRECATED_MESSAGE("Unused and not really useful")
+        size_type GetMaxContiguousAllocationSize() const
+        {
+            return 0;
+        }
+
+        /**
+         * Returns memory allocated by the allocator and available to the user for allocations.
+         * IMPORTANT: this is not the overhead memory this is just the memory that is allocated, but not used. Example: the pool allocators
+         * allocate in chunks. So we might be using one elements in that chunk and the rest is free/unallocated. This is the memory
+         * that will be reported.
+         */
+        //AZ_DEPRECATED_MESSAGE("Use GetFragmentedSize instead, this method was problematic because not all overhead memory is avaialble for allocations. "
```

Comment:
typo: avaialble -> available
```suggestion
        //AZ_DEPRECATED_MESSAGE("Use GetFragmentedSize instead, this method was problematic because not all overhead memory is available for allocations. "
```

Reply:
Fixed.